runtime/metricsview/executor.go

```diff
@@ -65,10 +67,46 @@ func (e *Executor) Close() {
 	e.olapRelease()
 }
 
-// Cacheable returns whether the result of running the given query is cacheable.
-func (e *Executor) Cacheable(qry *Query) bool {
-	// TODO: Get from OLAP instead of hardcoding
-	return e.olap.Dialect() == drivers.DialectDuckDB
+// CacheKey returns a cache key based on the executor's metrics view's cache key configuration.
+// If ok is false, caching is disabled for the metrics view.
+func (e *Executor) CacheKey(ctx context.Context) ([]byte, bool, error) {
+	spec := e.metricsView
+	// Cache is disabled for metrics views based on external table
+	if (spec.CacheEnabled != nil && !*spec.CacheEnabled) || (spec.CacheEnabled == nil && e.streaming) {
+		return nil, false, nil
+	}
+
+	if spec.CacheKeySql == "" {
+		if !e.streaming {
+			// for metrics views on rill managed tables no cache key specific to the metrics view is required
+			return []byte(""), true, nil
+		}
+		// watermark is the default cache key for streaming metrics views
+		watermark, err := e.loadWatermark(ctx, nil)
+		if err != nil {
+			return nil, false, err
+		}
+		return []byte(watermark.Format(time.RFC3339)), true, nil
+	}
+
+	res, err := e.olap.Execute(ctx, &drivers.Statement{
+		Query:    spec.CacheKeySql,
+		Priority: e.priority,
+	})
+	if err != nil {
+		return nil, false, err
+	}
+	defer res.Close()
+	var key string
+	for res.Next() {
+		if err := res.Scan(&key); err != nil {
+			return nil, false, err
+		}
+	}
```

Comment:
If the cache key SQL doesn't have an aggregation or `LIMIT`, this may accidentally process a ton of data. Maybe using `if` instead would provide at least a little protection against that?

```suggestion
	if res.Next() {
		if err := res.Scan(&key); err != nil {
			return nil, false, err
		}
	}
```